localstack-core/localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -756,8 +753,10 @@ def _resolve_refs_recursively(
                     {inner_list[0]: inner_list[1]},
                 )
 
+        # remove _aws_no_value_ from resulting references
+        clean_list = []
         for i in range(len(value)):
```

Comment:
nit: this no longer needs to be a list over the indices

```suggestion
        for item in value:
```